approach_data_structure_task.md

```diff
@@ -0,0 +1,168 @@
+# How to approach a new task, and its underlying data structure 
+
+When approaching a new task, it may seem overwhelming because you don’t know from where to start.
+
+In this chapter we will present a method that includes a few steps of logical thinking to help you find the starting point. The task is understanding the set of requirements for implementation and it is done by first finding the underlying data structure.
+
+A task is often composed of many small parts. We need to break the "problem" down to atom parts and reveal the entities and the relationships between them. It is ultimately a world of circles, lines, and arrows.
+
+Take a look at the following simple scenario: 
+
+
+> Authors can write articles.
+
+
+Let's try to identify the entities and relationships in this scenario. An amazingly quick and simple way to do it is to draw it by hand. Use circles for entities and lines for the relationships between them.
+
+![](111.jpg)
+
+That wasn't so hard, was it? In this case, the entities are `author` and `article`.  The line indicates that they are somehow related.
+
+A relationship defines how entities relate, so we can use arrows to express the reference between the entities. In order to know the direction of the arrow, we should ask two important questions:
+
+**Question 1:** Can the author write more than one article?
+
+Answer: Yes, she can! So, let's represent that in the sketch. It would look like this:
+
+![](2.jpg)
+
+And, if our author is particularly diligent, she can even write a million articles (theoretically of course), so this brings us to the next important question:
+
+**Question 2. The Million Question** Can and should a single `author` entity refer to a million `article` entities?
+
+Answer: No, the answer is definitely not. It would be a bad idea. 
+
+To understand this, we need to talk about the **meaning of reference**. When object A refers to object B it means object A “knows” about object B. This "knowing" becomes part of the information that object A holds.  So for example when we call object A from the database, it might also retrieve the information of all the "B" objects that it refers to. So, if theoretically there can be a million "B" objects, the system would retrieve them all. This is a very heavy task for the system that requires a lot of memory resources. We want to avoid this.
```

Comment:
Good! Let's keep the "A" and "B" explanation, and add your sentence after it, so it will do the link to the case we talk about.